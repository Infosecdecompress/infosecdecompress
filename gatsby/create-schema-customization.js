const createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions;
  const typeDefs = `
    type MarkdownRemark implements Node {
      frontmatter: Frontmatter
    }
    type SocialImage {
      publicURL: String
    }
    type Frontmatter {
      socialImage: SocialImage
    }
  `;

  createTypes(typeDefs);
};

module.exports = createSchemaCustomization;