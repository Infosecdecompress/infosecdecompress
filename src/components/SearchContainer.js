import React, { Component } from "react"
import Axios from "axios"
import * as JsSearch from "js-search"

class Search extends Component {
  state = {
    itemList: [],
    search: [],
    searchResults: [],
    isLoading: true,
    isError: false,
    searchQuery: "",
  }
  /**
   * React lifecycle method to fetch the data
   */
  async componentDidMount() {
    Axios.get("/feed-1.json")
      .then(result => {
        const feedData = result.data
        this.setState({ itemList: feedData.items })
        this.rebuildIndex()
      })
      .catch(err => {
        this.setState({ isError: true })
        console.log("====================================")
        console.log(`Something bad happened while fetching the data\n${err}`)
        console.log("====================================")
      })
  }

  /**
   * rebuilds the overall index based on the options
   */
  rebuildIndex = () => {
    var REGEX = /\s+/; // Split on spaces
    const tokenizer = {
      tokenize(text) {
        return text
          .split(REGEX)
          .filter(
            (text) => text // Filter empty tokens
          );
      }
    }
    const { itemList } = this.state
    const dataToSearch = new JsSearch.Search("id")
    dataToSearch.tokenizer = tokenizer
    /**PrefixIndexStrategy   AllSubstringsIndexStrategy ExactWordIndexStrategy  */
    dataToSearch.indexStrategy = new JsSearch.PrefixIndexStrategy()
    dataToSearch.sanitizer = new JsSearch.LowerCaseSanitizer()
    /**defines the search index  https://github.com/bvaughn/js-search#configuring-the-search-index*/
    dataToSearch.searchIndex = new JsSearch.TfIdfSearchIndex("title")
    dataToSearch.addIndex("title") // sets the index attribute for the data
    dataToSearch.addIndex("content") // sets the index attribute for the data
    dataToSearch.addDocuments(itemList) // adds the data to be searched
    this.setState({ search: dataToSearch, isLoading: false })
  }

  /**
   * handles the input change and perform a search with js-search
   * in which the results will be added to the state
   */
  searchData = e => {
    const { search } = this.state
    const queryResult = search.search(e.target.value)
    this.setState({ searchQuery: e.target.value, searchResults: queryResult })
    var status = document.getElementById("result");
    if (e.target.value != "") {
      status.style.display = "block";
    }
    else {
      status.style.display = "none";
    }
  }
  handleSubmit = e => {
    e.preventDefault()
  }

  render() {
    const { itemList, searchResults, searchQuery } = this.state
    const queryResults = searchQuery === "" ? itemList : searchResults
    return (
      <div style={{margin:"5px auto"}}>
        <div style={{ margin: "0 auto" }}>
          <form onSubmit={this.handleSubmit}>
            <div style={{ margin: "0 auto" }}>
              <input
                id="Search"
                value={searchQuery}
                onChange={(evt) => this.searchData(evt)}
                placeholder="搜尋文章"
                style={{ 
                  margin: "0 auto", 
                  width: "250px",
                  padding: "5px 5px",
                  display: "inline-block",
                  border: "1px solid #ccc",
                  borderRadius: "4px",
                  boxSizing: "border-box",
                }}
              />
            </div>
          </form>
          <div>
            <table id="result"
              style={{
                width: "250px",
                borderCollapse: "collapse",
                borderRadius: "4px",
                border: "1px solid #ccc",
                display: "none",
              }}
            >
              <tbody>
                {queryResults.map(item => {
                  return (
                    <tr key={`row_${item.id}`}>
                      <td
                        style={{
                          fontSize: "14px",
                        }}
                      >
                        <a href={item.url}>
                          {item.title}
                        </a>
                      </td>
                    </tr>
                  )
                })}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    )
  }
}
export default Search