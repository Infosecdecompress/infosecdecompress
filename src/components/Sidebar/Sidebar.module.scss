@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.sidebar {
  width: 100%;

  .inner {
    @include padding(1, 0.75, 0);
    position: relative;
  }
}

@include breakpoint-sm {
  .sidebar {
    lost-column: 5/12;

    .inner {
      @include padding(1.25, 0.75, 0);

      &:after {
        background: $color-gray-border;
        background: linear-gradient(
          to bottom,
          $color-gray-border 0%,
          $color-gray-border 48%,
          $color-white 100%
        );
        bottom: 0;
        content: "";
        height: 540px;
        position: absolute;
        right: -10px;
        top: 30px;
        width: 1px;
      }
    }
  }
}

@include breakpoint-md {
  .sidebar {
    lost-column: 1/3;

    .inner {
      @include padding-equal(1.5);
    }
  }
}
