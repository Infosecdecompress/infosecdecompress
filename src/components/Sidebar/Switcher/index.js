// @flow strict
export { default } from './Switcher';
