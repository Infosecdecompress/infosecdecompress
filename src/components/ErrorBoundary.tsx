import React from "react"

import { ErrorBoundary } from "@sentry/react"
import { Link } from "gatsby"

const Fallback = ({ error, resetError }) => (
  <div role="alert">
    <p>Something went wrong</p>
    <pre>{error.message}</pre>
    <Link to="/" onClick={resetError}><a>Try again</a></Link>
  </div>
)
const ErrorBoundaryContainer = ({ children }) => <ErrorBoundary fallback={Fallback}>{children}</ErrorBoundary>

export default ErrorBoundaryContainer