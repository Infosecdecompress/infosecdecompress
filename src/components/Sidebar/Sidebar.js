// @flow strict
import React from 'react';
import Author from './Author';
import Contacts from './Contacts';
import Copyright from './Copyright';
import Menu from './Menu';
import styles from './Sidebar.module.scss';
import { useSiteMetadata } from '../../hooks';
import Search from "../SearchContainer";
import Switcher from './Switcher';

const siteConfig = require('../../../config');

type Props = {
  isIndex?: boolean,
};

const Sidebar = ({ isIndex }: Props) => {
  const { author, copyright, menu } = useSiteMetadata();
  const { darkFunctionalityIsOn } = siteConfig;

  return (
    <div className={styles['sidebar']}>
      <div className={styles['sidebar__inner']}>
        <Author author={author} isIndex={isIndex} />
        <Menu menu={menu} />
        <Contacts contacts={author.contacts} />
        <Search />
        {darkFunctionalityIsOn ? <Switcher /> : null}
        <Copyright copyright={copyright} />
        </div>
      </div>
  );
};

export default Sidebar;
