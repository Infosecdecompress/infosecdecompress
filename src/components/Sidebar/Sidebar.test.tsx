import React from "react";
import renderer from "react-test-renderer";

import { StaticQuery, useStaticQuery } from "gatsby";

import { Sidebar } from "@/components/Sidebar";
import * as mocks from "@/mocks";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("Sidebar", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  it("renders correctly", () => {
    const props = { isIndex: true };
    const tree = renderer.create(<Sidebar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
