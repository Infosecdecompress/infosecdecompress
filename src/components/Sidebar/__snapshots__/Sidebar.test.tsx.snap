// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div>
  <div>
    <div>
      <a
        href="/"
      />
      <div>
        <h1>
          <a
            href="/"
          >
            John Doe
          </a>
        </h1>
      </div>
      <p>
        InfosecDecompress.
      </p>
    </div>
    <nav>
      <ul>
        <li>
          <a
            href="/"
          >
            Articles
          </a>
        </li>
        <li>
          <a
            href="/pages/about"
          >
            About Me
          </a>
        </li>
        <li>
          <a
            href="/pages/contacts"
          >
            Contact Me
          </a>
        </li>
      </ul>
    </nav>
    <div
      style={
        {
          "margin": "5px auto",
        }
      }
    >
      <div
        style={
          {
            "margin": "0 auto",
          }
        }
      >
        <form
          onSubmit={[Function]}
        >
          <div
            style={
              {
                "margin": "0 auto",
              }
            }
          >
            <input
              id="Search"
              onChange={[Function]}
              placeholder="搜尋文章"
              style={
                {
                  "border": "1px solid #ccc",
                  "borderRadius": "4px",
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "margin": "0 auto",
                  "padding": "5px 5px",
                  "width": "250px",
                }
              }
              value=""
            />
          </div>
        </form>
        <div>
          <table
            id="result"
            style={
              {
                "border": "1px solid #ccc",
                "borderCollapse": "collapse",
                "borderRadius": "4px",
                "display": "none",
                "width": "250px",
              }
            }
          >
            <tbody />
          </table>
        </div>
      </div>
    </div>
    <div>
      Content is licensed under CC BY-SA 4.0
    </div>
  </div>
</div>
`;
